src/libasr/pass/pass_manager.h

```diff
@@ -77,18 +80,32 @@ namespace LCompilers {
 
         bool is_fast;
         bool apply_default_passes;
+        bool do_print_after_asr_pass;
 
         void _apply_passes(Allocator& al, LFortran::ASR::TranslationUnit_t* asr,
                            std::vector<std::string>& passes, PassOptions pass_options) {
             pass_options.runtime_library_dir = LFortran::get_runtime_library_dir();
+            if (do_print_after_asr_pass) {
+                std::string cmd = "mkdir asr_pass_log";
+                int err = system(cmd.c_str());
+                if (err) {
+                    std::cout << "The command '" + cmd + "' failed." << std::endl;
+                    exit(1);
+                }
+                LFortran::write_asr_to_file("./asr_pass_log/0_raw.log", *asr, false, false);
```

Comment:
This file is only available in LPython (in this PR), so it will fail in LFortran. We cannot use any LPython specific things in libasr. We'll have to figure out another solution to fix this.